### ISSUE TITLE
[Bug]: portenf.sa.gov.au Source Issue

### ISSUE THREAD

**Issue by illuzn** [mampfes/hacs_waste_collection_schedule#2218]
### I Have A Problem With:

A specific source

### What's Your Problem

Using the master branch up to commit [9739594](https://github.com/mampfes/hacs_waste_collection_schedule/commit/9739594e1db322bd616a2b485884de324f3147cb).

Source fetch fails.

At time of error, this is what the source looks like:

![image](https://github.com/mampfes/hacs_waste_collection_schedule/assets/57167030/df76f28c-a85b-49f1-babb-4701281d3aea)

### Source (if relevant)

portenf_sa_gov_au

### Logs

```Shell
fetch failed for source Port Adelaide Enfield, South Australia: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 145, in fetch entries = self._source.fetch() ^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/portenf_sa_gov_au.py", line 159, in fetch to_month = cal_header.split("-")[1].strip().split(" ")[0] ~~~~~~~~~~~~~~~~~~~~~^^^ IndexError: list index out of range
```

### Relevant Configuration

```YAML
Not Relevant
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by illuzn** [comment]
Issue appears to arise here:
https://github.com/mampfes/hacs_waste_collection_schedule/blob/9739594e1db322bd616a2b485884de324f3147cb/custom_components/waste_collection_schedule/waste_collection_schedule/source/portenf_sa_gov_au.py#L156-L160
Ordinarily, month will be a month span e.g. June-July 2024 however for the first week of the month there is no such span.

**Comment by 5ila5** [comment]
Fixed in #2222 
This will be part of the next release or is available now as master (git) version (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)

**Comment by illuzn** [comment]
Confirmed working. Thank you
